spec/lang/representation.md

```diff
@@ -123,19 +125,72 @@ fn encode_ptr<M: Memory>(ptr: ThinPointer<M::Provenance>) -> List<AbstractByte<M
     
 }
 
+impl PtrType {
+    // Returns a 2-tuple type representing this wide pointer or `None` if it is thin.
+    fn as_tuple_ty<M: Memory>(self) -> Option<Type> {
+        let meta_ty = match self.meta_kind() {
+            PointerMetaKind::None => throw!(),
```

Comment:
We *could* put a unit type here and then treat thin and wide pointers more uniformly?

Reply:
Maybe, but keep in mind that there needs to be a separate "base case" for thin pointers anyway, because the Pointer::decode will be called by the Tuple::decode, meaning it is recursive. Preventing this by an explicit panic seemed like the least error prone solution to me.

Reply:
Ah, good point. That should be noted in a comment somewhere.

But also, maybe there is a way here to structure this that does not require any unwraps -- something more like `if let Some(...)` then `if ...`?